test/bdd/features/introduce.feature

```diff
@@ -10,284 +10,177 @@ Feature: Introduce protocol
 
   @proposal_stop_unusual
   Scenario: Carol does not want the introduction
     Given   "Bob,Carol" exchange DIDs with "Alice"
 
     When   "Alice" sends introduce proposal to the "Bob" and "Carol"
-      And   "Bob" wants to know "Carol" and sends introduce response with approve
-      And   "Alice" continue with the introduce protocol
+    And   "Bob" wants to know "Carol" and sends introduce response with approve
+    And   "Carol" doesn't want to know "Bob" and sends introduce response
 
-    When   "Carol" doesn't want to know "Bob" and sends introduce response
-      And   "Alice" continue with the introduce protocol
-
-    Then   "Alice" checks the history of introduce protocol events "arranging,arranging,arranging,arranging,delivering,delivering,abandoning,abandoning,done,done"
-      And   "Bob" checks the history of introduce protocol events "deciding,deciding,waiting,waiting,abandoning,abandoning,done,done"
-      And   "Carol" checks the history of introduce protocol events "deciding,deciding,abandoning,abandoning,done,done"
+    Then   "Alice" checks the history of introduce protocol events "arranging,arranging,arranging,arranging,arranging,arranging,arranging,arranging,abandoning,abandoning,done,done"
+    And   "Bob" checks the history of introduce protocol events "deciding,deciding,waiting,waiting,abandoning,abandoning,done,done"
+    And   "Carol" checks the history of introduce protocol events "deciding,deciding,abandoning,abandoning,done,done" and stop
 
   @proposal_stop_unusual_with_request
   Scenario: Carol does not want the introduction the protocol starts with request
     Given   "Bob,Carol" exchange DIDs with "Alice"
 
     When   "Bob" sends introduce request to the "Alice" asking about "Carol"
-      And   "Alice" sends introduce proposal back to the "Bob" and requested introduce
-      And   "Bob" wants to know "Carol" and sends introduce response with approve
-
-    When   "Alice" continue with the introduce protocol
-      And   "Carol" doesn't want to know "Bob" and sends introduce response
-      And   "Alice" continue with the introduce protocol
+    Then   "Alice" sends introduce proposal back to the "Bob" and requested introduce
+    And   "Bob" wants to know "Carol" and sends introduce response with approve
+    And   "Carol" doesn't want to know "Bob" and sends introduce response
 
-    Then   "Alice" checks the history of introduce protocol events "arranging,arranging,arranging,arranging,delivering,delivering,abandoning,abandoning,done,done"
+    Then   "Alice" checks the history of introduce protocol events "arranging,arranging,arranging,arranging,arranging,arranging,abandoning,abandoning,done,done"
     And   "Bob" checks the history of introduce protocol events "requesting,requesting,deciding,deciding,waiting,waiting,abandoning,abandoning,done,done"
-    And   "Carol" checks the history of introduce protocol events "deciding,deciding,abandoning,abandoning,done,done"
-
-  @proposal_introducer_stop
```

Comment:
is this case no longer relevant?

Reply:
No, since I have deleted the action event after receiving a response it is not possible anymore.